contracts/oracle/src/state/supported_pairs.rs

```diff
@@ -194,23 +198,25 @@ where
         ))
     }
 
-    pub fn swap_pairs_df(&self) -> impl Iterator<Item = SwapLeg> + '_ {
```

Comment:
let's not bind the API to concrete types, why do we switch out from `impl Xxx` ???